core/extensions.rs

```diff
@@ -608,54 +566,49 @@ impl ExtensionBuilder {
 /// - "ext:my_extension/js/01_hello.js"
 /// - "ext:my_extension/js/02_goodbye.js"
 /// ```
-#[cfg(not(feature = "include_js_files_for_snapshotting"))]
+#[cfg(not(all(
+  feature = "exclude_js_sources",
+  not(feature = "force_include_js_sources")
+)))]
 #[macro_export]
 macro_rules! include_js_files {
-  ($name:ident dir $dir:literal, $($file:literal,)+) => {
-    vec![
-      $($crate::ExtensionFileSource {
-        specifier: concat!("ext:", stringify!($name), "/", $file),
-        code: $crate::ExtensionFileSourceCode::IncludedInBinary(
-          include_str!(concat!($dir, "/", $file)
-        )),
-      },)+
-    ]
+  ($name:ident $(force_include_js_sources $($dummy:block)?,)? $(dir $dir:literal,)? $($file:literal,)*) => {
+    $crate::force_include_js_files!($name $(dir $dir,)? $($file,)*)
   };
+}
 
-  ($name:ident $($file:literal,)+) => {
-    vec![
-      $($crate::ExtensionFileSource {
-        specifier: concat!("ext:", stringify!($name), "/", $file),
-        code: $crate::ExtensionFileSourceCode::IncludedInBinary(
-          include_str!($file)
-        ),
-      },)+
-    ]
+#[cfg(all(
+  feature = "exclude_js_sources",
+  not(feature = "force_include_js_sources")
+))]
+#[macro_export]
+macro_rules! include_js_files {
+  ($name:ident force_include_js_sources $($dummy:block)?, $(dir $dir:literal,)? $($file:literal,)*) => {
+    $crate::force_include_js_files!($name $(dir $dir,)? $($file,)*)
+  };
+
+  ($name:ident $(dir $dir:literal,)? $($file:literal,)*) => {
+    vec![]
   };
 }
 
-#[cfg(feature = "include_js_files_for_snapshotting")]
 #[macro_export]
-macro_rules! include_js_files {
-  ($name:ident dir $dir:literal, $($file:literal,)+) => {
+macro_rules! force_include_js_files {
+  ($name:ident dir $dir:literal, $($file:literal,)*) => {
     vec![
       $($crate::ExtensionFileSource {
         specifier: concat!("ext:", stringify!($name), "/", $file),
-        code: $crate::ExtensionFileSourceCode::LoadedFromFsDuringSnapshot(
-          std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR")).join($dir).join($file)
-        ),
-      },)+
+        code: include_str!(concat!($dir, "/", $file)),
+      },)*
     ]
   };
 
-  ($name:ident $($file:literal,)+) => {
+  ($name:ident $($file:literal,)*) => {
     vec![
       $($crate::ExtensionFileSource {
         specifier: concat!("ext:", stringify!($name), "/", $file),
-        code: $crate::ExtensionFileSourceCode::LoadedFromFsDuringSnapshot(
-          std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR")).join($file)
-        ),
-      },)+
+        code: include_str!($file),
```

Comment:
This will cause the source code to be included in the resulting binary even though it's snapshotted. Which is visible in the benchmark page https://deno.com/benchmarks for binary size

Reply:
It's supposed to work because no `ExtensionFileSource`s are registered in the final binary in the first place, they are replaced by an empty vec in `include_js_files!()`. I debug printed the esm files for extension builders after the `ext.esm()` call and they were all empty. But something apparently went wrong...

One difference this PR makes is that the static sources are included in the compilation of the `build.rs` scripts, where as before they were loaded at runtime of the `build.rs`. That shouldn't make a difference should it?

Reply:
> One difference this PR makes is that the static sources are included in the compilation of the build.rs scripts, where as before they were loaded at runtime of the build.rs. That shouldn't make a difference should it?

I would assume that would be the case, I can't say for sure off the top of my head.